examples/score_based_providers/score_based_providers.py

```diff
@@ -131,4 +132,13 @@ async def main() -> None:
 
 
 if __name__ == "__main__":
-    asyncio.run(main())
+    loop = asyncio.get_event_loop()
+    task = loop.create_task(main())
+    try:
+        loop.run_until_complete(task)
+    except KeyboardInterrupt:
+        task.cancel()
+        try:
+            loop.run_until_complete(task)
+        except asyncio.CancelledError:
+            pass
```

Comment:
```suggestion
    asyncio.run(main())
```

Same as previous similar block.